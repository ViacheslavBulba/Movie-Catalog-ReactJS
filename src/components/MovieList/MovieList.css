.movie-list-container {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    flex-wrap: wrap;
    margin-left: 30px;
    margin-right: 30px;
    min-height: 400px;
}

.no-movies-text {
    font-family: 'Source Sans Pro';
    font-size: 48px;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
}
