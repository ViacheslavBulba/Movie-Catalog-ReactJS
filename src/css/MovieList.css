.movie-list-container {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    flex-wrap: wrap;
    margin-left: 30px;
    margin-right: 30px;
    min-height: 400px;
}
