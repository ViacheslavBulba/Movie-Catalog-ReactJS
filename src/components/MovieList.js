
import React from 'react';
import '../css/MovieList.css';
import MovieCard from './MovieCard';

export default function MovieList({ listInput }) {

    return (
        <div className="movie-list-container">
            {listInput.map(item => (
                <MovieCard movie={item} key={item.id} />
            ))}
        </div>
    )
}
