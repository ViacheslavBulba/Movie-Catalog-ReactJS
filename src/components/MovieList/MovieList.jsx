import React from 'react';
import './MovieList.css';
import MovieCard from '../MovieCard/MovieCard';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

export default function MovieList(props) {
    const movies = useSelector((state) => state.movies);

    return (
        <div className='movie-list-container'>
            {movies.map((item) => (
                <MovieCard
                    movie={item}
                    key={item.id}
                    showOverview={props.showOverview}
                />
            ))}
        </div>
    );
}

MovieList.propTypes = {
    showOverview: PropTypes.func.isRequired,
};
