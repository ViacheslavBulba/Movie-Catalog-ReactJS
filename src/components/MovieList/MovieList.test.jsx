import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { MemoryRouter } from 'react-router';
import MovieList from './MovieList';

const mockStore = configureStore();

const initialState = {
  pending: false,
  movies: [
    {
      id: 1,
      title: 'mock movie 1',
      tagline: 'mock tagline 1',
      vote_average: 9.9,
      vote_count: 0,
      release_date: '2021-04-18',
      poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      overview: 'mock overview',
      budget: 1000,
      revenue: 2000,
      genres: ['Drama', 'Romance'],
      runtime: 90,
    },
    {
      id: 2,
      title: 'mock movie 2',
      tagline: 'mock tagline 2',
      vote_average: 9.9,
      vote_count: 0,
      release_date: '2021-01-01',
      poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      overview: 'mock overview 2',
      budget: 1000,
      revenue: 2000,
      genres: ['Drama', 'Romance'],
      runtime: 100,
    },
  ],
  error: null,
  sortBy: 'release_date',
  sortOrder: 'desc',
  filerByGenres: [],
  search: '',
  searchBy: 'title',
  movieToOverview: null,
};

const store = mockStore(initialState);

test('snapshot test for movie list', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Provider store={store}>
        <MovieList />
      </Provider>
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
