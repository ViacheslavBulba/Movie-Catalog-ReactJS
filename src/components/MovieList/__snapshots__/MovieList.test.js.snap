// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for movie list 1`] = `
<div
  className="movie-list-container"
>
  <span
    className="no-movies-text"
  >
    No Movies Found
  </span>
</div>
`;
